### ISSUE TITLE
Stability of submissions in form group

### ISSUE THREAD

**Issue by Aryllen** [Sage-Bionetworks/stopadforms#39]
This is not necessarily an issue with the stopadforms application, but it may be beneficial to put a note in the application depending on what's discovered regarding this problem. I am documenting it here in case this is an ongoing issue or if we should add something to the application due to it.

The problem: At one point, I rejected one of my submissions. I cannot remember the exact one, but the rejection had the desired affect: 
- the submission was filtered out when displaying submissions currently waiting for review, 
- the submission appeared in the rejected set of submissions with the reason (internally, not in either application), and 
- the web application stated that the submission had been rejected.

When I checked earlier, I have no rejected submissions in the form group. Since I cannot remember which of the submissions I originally rejected, I am not sure if the submission changed status back to SUBMITTED_WAITING_FOR_REVIEW or if it was removed from the database (something that I was under the impression could not be done for some reason).

Overall, this is not a huge problem for the moment, but could be an issue in the future depending on what happened.

If the submission changed status back to SUBMITTED_WAITING_FOR_REVIEW (not likely), then this would be a fairly large issue for the clients, especially given the number of test submissions we would want to reject before handing over access. It would be frustrating for the clients to have to re-reject submissions.

If the submission was removed from the database since it was rejected and no longer needed, then this is less of a problem. The issue here would be the time-frame for when the database was purged of rejected submissions. If this is based on time since rejection, then the submitter would have some time to check back on the status of their submission. If this is based on a date, such as the first of the month, then the submitter may not have time to check their status between rejection and purge.

Additionally, it is unknown if accepted submissions have this same problem since there have been none, thus far.

 As of now, I have rejected formDataId 98, another of my test submissions, to see what happens, and am documenting the exact one here in case my memory fails me, again.

**Comment by Aryllen** [comment]
Update:

Spoke with John Hill. The database does not remove submissions automatically. Missing submission is odd and may be a fluke. Both my submission, 98, and Kara's submission, 209, have been rejected and will see what happens to those.

On a great note, if we give John a list of submissions, he can delete them from the database. This could be done to clear out the extra nonsense before passing off to the client.

**Comment by Aryllen** [comment]
Update: Kara's submission is still rejected. My submission, 98, seems to have disappeared, but I think I might know (a little) of what is going on.

I deleted my submissions (except Cheese-Wiz, which was requested to be left in for now). We asked to delete the test submissions from the database. This is not possible unless the original owner of the submission deletes the form. Instead, we rejected the test submissions. However, within a week, all rejected submissions except for Kara's #209 were back in the `SUBMITTED_WAITING_FOR_REVIEW` status. My guess for why my submission appeared to disappear was that I didn't realize that a `SUBMITTED_WAITING_FOR_REVIEW` submission that I was deleting was the same submission I had previously rejected.

In short, for some reason, all except one rejected submission changed their status back to not being rejected. A Jira will need to be filed for this.

**Comment by karawoo** [comment]
@Aryllen do I remember correctly that this has been resolved?

**Comment by Aryllen** [comment]
Yes, this is fixed and we can close this.
